fortran/src/H5Aff.F90

```diff
@@ -200,6 +200,82 @@ END FUNCTION H5Acreate2
 
   END SUBROUTINE h5acreate_f
 
+!
+!****s* H5A/h5acreate_async_f
+!
+! NAME
+!  h5acreate_async_f
+!
+! PURPOSE
+!  Creates a dataset as an attribute of a group, dataset, or named datatype
+!
+! INPUTS
+!  loc_id 	 - identifier of an object (group, dataset,
+!                  or named datatype) attribute is attached to
+!  name 	 - attribute name
+!  type_id 	 - attribute datatype identifier
+!  space_id 	 - attribute dataspace identifier
+!  es_id         - event stack identifier
+!
+! OUTPUTS
+!  attr_id 	 - attribute identifier
+!  hdferr 	 - Returns 0 if successful and -1 if fails
+! OPTIONAL PARAMETERS
+!  acpl_id 	 - Attribute creation property list identifier
+!  appl_id 	 - Attribute access property list identifier
+!
+! AUTHOR
+!  M. Breitenfeld
+!  Jan 27 2021
+!
+! SOURCE
+  SUBROUTINE h5acreate_aysnc_f(loc_id, name, type_id, space_id, attr_id, &
+       es_id, hdferr, acpl_id, aapl_id)
+    IMPLICIT NONE
+    INTEGER(HID_T), INTENT(IN) :: loc_id   ! Object identifier
+    CHARACTER(LEN=*), INTENT(IN) :: name   ! Attribute name
+    INTEGER(HID_T), INTENT(IN) :: type_id  ! Attribute datatype identifier
+    INTEGER(HID_T), INTENT(IN) :: space_id ! Attribute dataspace identifier
+    INTEGER(HID_T), INTENT(OUT) :: attr_id ! Attribute identifier
+    INTEGER(HID_T), INTENT(IN) :: es_id    ! Attribute identifier
+    INTEGER, INTENT(OUT) :: hdferr         ! Error code:
+                                           ! 0 on success and -1 on failure
+!*****
+    INTEGER(HID_T), OPTIONAL, INTENT(IN) :: acpl_id ! Attribute creation property list identifier
+    INTEGER(HID_T), OPTIONAL, INTENT(IN) :: aapl_id ! Attribute access property list identifier
+
+    INTEGER(HID_T) :: acpl_id_default
+    INTEGER(HID_T) :: aapl_id_default
+    CHARACTER(LEN=LEN_TRIM(name)+1,KIND=C_CHAR) :: c_name
+    INTERFACE
+       INTEGER(HID_T) FUNCTION H5Acreate_async(loc_id, name, type_id, &
```

Comment:
I also had that concern about async, but I'm assuming they are here to stay and can not be compiled out.  If we don't think that is the case, and we want to future-proof it, then I will have to change it to use a common code like Quincey.

I don't think there is much overhead, I am more concerned about what the overhead is of calling C_async APIs instead of calling sync API directly. Maybe that is a better case for using a common subroutine instead?

Yes, we will need to always include the 3 extra parameters when calling the C APIs.